src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl

```diff
@@ -1,6 +1,68 @@
 <policies>
     <inbound>
+      <!-- Extract payment method name for create redirectUrl -->
       <set-variable  name="walletToken"  value="@(context.Request.Headers.GetValueOrDefault("Authorization", "").Replace("Bearer ",""))"  />
+      <set-variable name="requestBody" value="@(context.Request.Body.As<JObject>(preserveContent: true))" />
+      <set-variable name="paymentMethodId" value="@((string)((JObject) context.Variables["requestBody"])["paymentMethodId"])" />
```

Comment:
I would move this api call to payment-methods after pm `/start-session`.
In case of invalid input wallet token we will avoid perform an useless api call to payment method service. what do you think?

Reply:
Fixed with eccca1db4601c4ce53d0221144728206b4135e9f